packages/core/src/components/section/section.tsx

```diff
@@ -110,6 +111,11 @@ export interface SectionProps extends Props, Omit<HTMLDivProps, "title">, React.
      * Note that the header will only be rendered if `title` is provided.
      */
     title?: JSX.Element | string;
+
+    /**
+     * @default H6
+     */
+    titleTagName?: keyof JSX.IntrinsicElements | React.FC<React.AllHTMLAttributes<HTMLElement>>;
```

Comment:
can we limit this to only `HTMLHeadingElement` function components, so that users are encouraged to use Blueprint's heading components? also I think the JSDoc needs to be improved

```suggestion
    /**
     * The heading component to use when rendering the title, typically one of `H1`, `H2,
     * `H3`, `H4`, `H5`, or `H6` from `@blueprintjs/core`.
     *
     * @default H6
     */
    titleHeadingComponent?: React.FC<React.AllHTMLAttributes<HTMLHeadingElement>>;
```

Reply:
Actually, the purpose for this change, four our application at least, is specifically to _not_ have it be inside of a heading element. This is so that we can pass an element to the `title` prop that has a heading element _within_ it. In our case, we are doing something like:
```javascript
title={
      <div>
        <H2 id={titleId}>{title}</H2>
        {titleRightElement}
      </div>
    }
```
so the goal here was to set `titleTagName` to `div`.

The other option I see here would be to implement `titleHeadingComponent` the way you said, and then add our `titleRightElement`. Thoughts?

Reply:
Third option: we scrap this change and add a `titleProps` prop which gets spread to the `H6`

Reply:
Ok, how about a render prop API instead?


```ts
type SectionTitleProps = Pick<React.HTMLAttributes<HTMLElement>, "className" | "id">;
```

```ts
/**
 * Optional title renderer function. It is recommended to render a Blueprint `<H6>` element as part of the title.
 *
 * @default H6
 */
renderTitle?: React.FC<SectionTitleProps>;
```

Note: this type is informed by a recent fix we had to make for a similar render prop API in https://github.com/palantir/blueprint/pull/6521

Reply:
Alright I take back my last comment, your first solution would work. Kept that solution. Renamed to `renderTitle` instead of `titleHeadingComponent` though. Up for whatever you want.

Reply:
upon further review, I decided that `titleRenderer` is a better name. I pushed a commit with that update